### ISSUE TITLE
API for long-lived processes and custom paths

### ISSUE THREAD

**Issue by kornelski** [rustsec/rustsec#663]
It seems that the API of `Repository` and `Database` assumes that I either use both default git URL and default disk location, or that I customize both. But I'd like to store the database on disk in a custom location, but still use the default URL.

Second issue about the API is that it doesn't support updating the `Database` after it's been created. I'm planning to keep the database open in a process with arbitrarily long uptime. `Database::fetch` and `Repository::fetch` require creating a new database instance, and have the all-or-nothing path customization dichotomy.

**Comment by tarcieri** [comment]
Yeah, there hasn't been a whole lot of thought put into those APIs. They were largely written to do what `cargo-audit` needed and not much else.
